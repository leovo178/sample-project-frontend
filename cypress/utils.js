/* eslint-disable no-undef */
const generateRandomNumber = (min, max) => {
	min = Math.ceil(min)
	max = Math.floor(max)
	return Math.floor(Math.random() * (max - min)) + min
}

const clickItemByTestId = testId =>
	cy
		.findByTestId(testId)
		.should('exist')
		.click()

const mockMessage = 'mockmessage'

export { generateRandomNumber, clickItemByTestId, mockMessage }
