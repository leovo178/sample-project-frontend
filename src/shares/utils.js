const AUTH_TOKEN = 'AUTH_TOKEN'

const getToken = () => localStorage.getItem(AUTH_TOKEN)
const setToken = token => localStorage.setItem(AUTH_TOKEN, token)
const deleteToken = () => localStorage.removeItem(AUTH_TOKEN)

const validateEmail = email => {
	const emailRegex = /^[a-z][a-z0-9_\.]{4,32}@[a-z0-9]{2,}(\.[a-z0-9]{2,4}){1,2}$/gm
	return email.match(emailRegex)
}

const validateName = name => {
	const nameRegex = /^[a-zA-Z0-9]{3,30}$/
	return name.match(nameRegex)
}

const validatePassword = password => {
	const passwordRegex = /^[a-zA-Z0-9]{3,}$/
	return password.match(passwordRegex)
}

export {
	getToken,
	setToken,
	deleteToken,
	validateEmail,
	validateName,
	validatePassword,
}
