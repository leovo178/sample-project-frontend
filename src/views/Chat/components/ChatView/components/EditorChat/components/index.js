export { default as SendReservationDialog } from './SendReservationDialog'
