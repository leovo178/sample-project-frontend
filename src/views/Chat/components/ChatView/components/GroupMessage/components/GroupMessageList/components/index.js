export { default as ImageMessage } from './ImageMessage'
export { default as LastCardIndicator } from './LastCardIndicator'
export { default as FileMessage } from './FileMessage'
export { default as TextMessage } from './TextMessage'
export { default as ReservationMessage } from './ReservationMessage'
