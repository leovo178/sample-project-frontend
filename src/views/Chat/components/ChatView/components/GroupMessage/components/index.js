export { default as GroupBreak } from './GroupBreak'
export { default as GroupMessageList } from './GroupMessageList'
