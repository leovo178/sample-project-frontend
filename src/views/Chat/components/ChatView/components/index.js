export { default as EditorChat } from './EditorChat'
export { default as GroupMessage } from './GroupMessage'
export { default as Header } from './Header'
export { default as MessageCard } from './MessageCard'
export { default as PickerEmoji } from './PickerEmoji'
export { default as ViewMessage } from './ViewMessage'
export { default as MessageReservation } from './MessageReservation'
