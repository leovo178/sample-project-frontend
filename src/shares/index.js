import { getToken, setToken, deleteToken } from './utils'

export { getToken, setToken, deleteToken }
