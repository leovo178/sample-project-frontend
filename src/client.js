import { ApolloClient } from 'apollo-client'
import { createUploadLink } from 'apollo-upload-client'
import { InMemoryCache } from 'apollo-cache-inmemory'
import { setContext } from 'apollo-link-context'

import { getToken } from './shares/utils'
import { APOLLO_SERVER } from './configs.local'

const typeDefs = {}

const resolvers = {
	Mutation: {
		setUserSearchValue: (_, { searchValue }, { cache }) => {
			cache.writeData({
				data: {
					userSearchValue: searchValue,
				},
			})
			return searchValue
		},
		setSelectedUser: (_, { selectedUser }, { cache }) => {
			cache.writeData({
				data: {
					selectedUser,
				},
			})
			return selectedUser
		},
		setUploadedFile: (_, { file }, { cache }) => {
			if (file) {
				cache.writeData({
					data: {
						file,
					},
				})
			}
			return file
		},
		setUserSearchValueOfMain: (_, { searchValue }, { cache }) => {
			cache.writeData({
				data: {
					userSearchValueOfMain: searchValue,
				},
			})
			return searchValue
		},
		setSelectedUserOfMain: (_, { selectedUser }, { cache }) => {
			cache.writeData({
				data: {
					selectedUserOfMain: selectedUser,
				},
			})
			return selectedUser
		},

		setMessageSearchValueOfMain: (_, { searchValue }, { cache }) => {
			cache.writeData({
				data: {
					messageSearchValueOfMessage: searchValue,
				},
			})
			return searchValue
		},
		setMessageCreateValueOfMain: (_, { createValue }, { cache }) => {
			cache.writeData({
				data: {
					messageCreateValueOfMessage: createValue,
				},
			})
			return createValue
		},
		resetCache: (_, { data }, { cache }) => {
			cache.writeData({
				data,
			})
			return data
		},
	},
}

const httpLink = createUploadLink({
	uri: APOLLO_SERVER,
	credentials: 'same-origin',
})

const authLink = setContext((_, { headers }) => {
	const token = getToken()
	return {
		headers: {
			...headers,
			authorization: token ? `Bearer ${token}` : '',
		},
	}
})

const cache = new InMemoryCache()

const client = new ApolloClient({
	cache,
	link: authLink.concat(httpLink),
	typeDefs,
	resolvers,
})

const initialState = {
	userSearchValue: '',
	messageCreateValueOfMessage: '',
	messageSearchValueOfMessage: '',
	userSearchValueOfMain: '',
	selectedUser: {
		id: '',
		name: '',
		email: '',
		__typename: 'User',
	},
	file: {
		filename: '',
		link: '',
		__typename: 'File',
	},
	selectedUserOfMain: {
		id: '',
		name: '',
		email: '',
		__typename: 'UserOfMain',
	},
}

cache.writeData({
	data: initialState,
})

export { client, initialState }
