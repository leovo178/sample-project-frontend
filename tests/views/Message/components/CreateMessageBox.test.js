import React from 'react'
import { act, cleanup, fireEvent } from '@testing-library/react'

import { createMockClient } from 'mock-apollo-client'
import { InMemoryCache } from 'apollo-cache-inmemory'

import { CreateMessageBox } from '@views/Message/components'

import { initialState, resolvers } from '@src/client'
import { renderWithApolloClient } from '@tests/shares/utils'

describe('CreateMessageBox', () => {
	const mockProps = {
		placeholder: 'text...',
		defaultValue: '',
		width: '350',
		onSubmit: jest.fn(),
		onChange: jest.fn(),
	}

	const cache = new InMemoryCache()
	cache.writeData({
		data: { ...initialState, messageCreateValueOfMessage: '' },
	})
	const mockClient = createMockClient({ cache, resolvers })

	afterEach(() => {
		cleanup()
	})

	it('should match snapshot', () => {
		const props = { ...mockProps }
		const { container } = renderWithApolloClient(
			<CreateMessageBox {...props} />,
			mockClient
		)

		expect(container).toMatchSnapshot()
	})

	it('should render a input with "text..." placeholder and a button with "Save" title', () => {
		let props = { ...mockProps }
		const { getByPlaceholderText, getByText } = renderWithApolloClient(
			<CreateMessageBox {...props} />,
			mockClient
		)

		expect(getByPlaceholderText(props.placeholder)).toBeTruthy()
		expect(getByText('Save')).toBeTruthy()
	})

	it('should call correctly onSubmit function when click "Save" button', () => {
		const props = { ...mockProps }
		const { getByText } = renderWithApolloClient(
			<CreateMessageBox {...props} />,
			mockClient
		)

		fireEvent.click(getByText('Save'))

		expect(props.onSubmit).toHaveBeenCalled()
	})

	it('should change value of input', async () => {
		const props = { ...mockProps }
		const { getByPlaceholderText } = renderWithApolloClient(
			<CreateMessageBox {...props} />,
			mockClient
		)
		const input = getByPlaceholderText(props.placeholder)

		expect(input.value).toBe('')

		await act(async () => {
			fireEvent.change(input, {
				target: { value: 'Hello world' },
			})
		})

		expect(input.value).toBe('Hello world')
	})

	it('should enable input when focus on input', async () => {
		const props = { ...mockProps }
		const { getByPlaceholderText } = renderWithApolloClient(
			<CreateMessageBox {...props} />,
			mockClient
		)
		const input = getByPlaceholderText(props.placeholder)

		act(() => {
			input.focus()
		})

		expect(input).toBe(document.activeElement)
	})
})
