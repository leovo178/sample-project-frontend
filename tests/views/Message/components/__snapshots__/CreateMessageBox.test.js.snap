// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateMessageBox should match snapshot 1`] = `
<div
  class="MuiBox-root MuiBox-root-7 makeStyles-root-1"
>
  <div
    class="MuiFormControl-root MuiTextField-root makeStyles-input-2 makeStyles-input-5"
  >
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiOutlinedInput-input"
        placeholder="text..."
        type="text"
        value=""
      />
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-46 MuiOutlinedInput-notchedOutline"
        style="padding-left: 8px;"
      >
        <legend
          class="PrivateNotchedOutline-legend-47"
          style="width: 0.01px;"
        >
          <span>
            â€‹
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-3 MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      Save
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
